src/test_workflow/integ_test/integ_test_runner.py

```diff
@@ -43,14 +43,13 @@ def run(self) -> TestSuiteResults:
                 if component.name in self.test_manifest.components:
                     test_config = self.test_manifest.components[component.name]
                     if test_config.integ_test:
-                        if 'ci-groups' in test_config.integ_test.keys():
+                        if self.args.ci_group:
                             orig_component_name = component.name
-                            for i in range(1, test_config.integ_test['ci-groups'] + 1):
-                                component.name = f"{orig_component_name}-ci-group-{i}"
-                                test_suite = self.__create_test_suite__(component, test_config, work_dir.path)
-                                test_results = test_suite.execute_tests()
-                                [self.test_recorder.test_results_logs.generate_component_yml(result_data) for result_data in test_suite.result_data]
-                                all_results.append(component.name, test_results)
+                            component.name = f"{orig_component_name}-ci-group-{self.args.ci_group}"
+                            test_suite = self.__create_test_suite__(component, test_config, work_dir.path)
+                            test_results = test_suite.execute_tests()
+                            [self.test_recorder.test_results_logs.generate_component_yml(result_data) for result_data in test_suite.result_data]
+                            all_results.append(component.name, test_results)
```

Comment:
Looks like if and else block has a duplicate code. Does the below recommendation makes sense? Only change component-name if ci-group is present. Everything goes on as before. 

```python
if self.args.ci_group:
    orig_component_name = component.name
    component.name = f"{orig_component_name}-ci-group-{self.args.ci_group}"

test_suite = self.__create_test_suite__(component, test_config, work_dir.path)
test_results = test_suite.execute_tests()
[self.test_recorder.test_results_logs.generate_component_yml(result_data) for result_data in test_suite.result_data]
all_results.append(component.name, test_results)